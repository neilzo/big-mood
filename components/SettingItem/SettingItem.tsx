import React from 'react';
import { StyleSheet, Text, View, TouchableHighlight } from 'react-native';

import colorVariables from '../colorVariables';
import Icon from '../Icon/Icon';

interface Props {
  title: string;
  icon: string;
  settingRoute: string;
  onPress: (routeName: string) => void;
}
const SettingItem = ({ title, icon, onPress, settingRoute }: Props) => {
  return (
    <TouchableHighlight
      onPress={() => onPress(settingRoute)}
      style={styles.touchWrap}
    >
      <View style={styles.container}>
        <View style={styles.iconWrap}>
          <Text>{icon}</Text>
        </View>
        <Text>{title}</Text>
        <View style={styles.arrowIcon}>
          <Icon name="chevron-right" size={20} />
        </View>
      </View>
    </TouchableHighlight>
  );
};

const styles = StyleSheet.create({
  touchWrap: {
    alignSelf: 'stretch',
  },
  container: {
    alignSelf: 'stretch',
    alignItems: 'center',
    flexDirection: 'row',
    borderTopWidth: 1,
    borderBottomWidth: 1,
    borderColor: colorVariables.borderColor,
    backgroundColor: colorVariables.white,
    padding: 10,
    marginTop: 15,
  },
  iconWrap: {
    marginRight: 10,
  },
  arrowIcon: {
    marginLeft: 'auto',
  },
});

export default SettingItem;
