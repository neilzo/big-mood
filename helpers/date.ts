import moment from 'moment';

const getPrettyDate = (date: Date) => moment(date).format('MMM Do YYYY');

const getPrettyTime = (date: Date) => moment(date).format('h:mma');

export default { getPrettyDate, getPrettyTime };
