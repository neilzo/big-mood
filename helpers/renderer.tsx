import 'react-native';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

export const createComponent = (component: any) => {
  return renderer.create(component);
};

// export const createComponentWithRedux = (children: any, props = {}) => {
//   return renderer.create(<Provider {...props}>{children}</Provider>);
// };
