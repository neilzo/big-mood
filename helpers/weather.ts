interface WeatherResults {
  weather: Array<{ main: string; icon: string }>;
  main: {
    temp: number;
  };
}

const kelvinToFarenheit = (kelvin: number) => {
  return Math.round(kelvin * (9 / 5) - 459.76);
};

const computeOverallWeather = (weatherResults: WeatherResults) => ({
  icon: weatherResults.weather[0].icon,
  description: weatherResults.weather[0].main,
  temperature: kelvinToFarenheit(weatherResults.main.temp),
});

export { kelvinToFarenheit, computeOverallWeather };
