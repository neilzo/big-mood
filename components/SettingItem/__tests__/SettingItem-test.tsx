import React from 'react';
import SettingItem from '../SettingItem';
import { createComponent } from '../../../helpers/renderer'; // todo resolve imports to avid that ugly thang

jest.mock('../../Icon/Icon', () => 'Icon');

const mockProps = {
  title: 'Terrace House',
  icon: 'chess-king',
  settingRoute: 'MoodSettings',
  onPress: () => {},
};

describe('SettingItem', () => {
  it('renders correctly', () => {
    expect(
      createComponent(<SettingItem {...mockProps} />).toJSON()
    ).toMatchSnapshot();
  });
});
