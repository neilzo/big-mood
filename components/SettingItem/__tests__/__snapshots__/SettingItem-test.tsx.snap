// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingItem renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "stretch",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#fff",
        "borderBottomWidth": 1,
        "borderColor": "#ccc",
        "borderTopWidth": 1,
        "flexDirection": "row",
        "marginTop": 15,
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "marginRight": 10,
        }
      }
    >
      <Text>
        chess-king
      </Text>
    </View>
    <Text>
      Terrace House
    </Text>
    <View
      style={
        Object {
          "marginLeft": "auto",
        }
      }
    >
      <Icon
        name="chevron-right"
        size={20}
      />
    </View>
  </View>
</View>
`;
